EIPS/eip-crosschain-id.md

```diff
@@ -0,0 +1,131 @@
+
+* RegisterBlockchainId
+* ModifyName
+* ModifyParameters
+* VerifyBlockchainId
+* QueryBlockChainId
+* LookupBlockchainId
+* QueryBlockchainOwner
+* QueryBlockchainName(Blockchain-id, NameType)
+* QueryParentBlockchainId
+
+## Rationale
+
+The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.
+With the success of Bitcoin and Ethereum, various blockchains such as EOS, Ripple, Litecoin, Besu, Wanchain and the like have been developed and are growing at a fast pace.  There are also other private and consortium blockchains such as Hyperledger Fabric, Hyperledger Besu, Stellar, Corda, Quorum that only allow nodes with permitted identities to join the blockchain network.  The growth of public and private blockchains imposes
+challenges for inter-chain interoperability, particularly when these chains are
+heterogeneous and incompatible.
+Enterprise Ethereum Alliance formed Crosschain Interoperability Task Force (CITF) to look into common crosschain problems and solutions. CITF team noticed that there is a lack of unique identifier to charaterise and describe a blockchain. Several proprosals were discussed in EEA Crosschain Interoperability Task Force meetings and discussions.  We have considered various alternative specifications such as using a random unique hex string to represent a blockchain.  The drawback of this method is that the random id can not be used to verify a blockchain's intrinsic identity such as the blockhash of the genesis block.  A second alternative is simply using a genesis blockhash to represent a blockchain id for crosschain operations.  The drawback of this is that this id does not have information about the property of the blockchain and it has problem when a blockchain is forked into two blockchain.  After various discussion, we finally came up with hybrid crosschain identity specification that allows a blockchain to be identified and verified both manually and through computer programs.
+
+## Backwards Compatibility
+
+Crosschainid can be backward compatible with EIP-155.  The crosschain id contains a 4 byte segment to record the chainid based on EIP-155.
+
+## Test Cases
+
+The test cases should cover the generation of crosschain id, the parsing of an crosschain id to ensure the various segments of the crosschain id meets the specification.
+For the crosschain registration smart contract, the test cases should cover the testing of all functions and also need to ensure that the security of each functions are properly implemented.
+
+## Implementation
```

Comment:
This section is generally being deprecated, so if you don't have an implementation to provide here inline I recommend removing it altogether.

Reply:
Sure.  Will remove this one.

Reply:
Discussed this in EEA Crosschain taskforce meeting.  Maybe leave this section for now as a place holder for discussions on implementations at the smart contract layer or core layer.